### ISSUE TITLE
disruptor should be optional=true, not scope=provided

### ISSUE THREAD

**Issue by bbottema** [vibur/vibur-object-pool#7]
Hi, I'm just going by the description in your maven POM:

```xml
<!-- Disruptor dependency will be used only if MultithreadConcurrentQueueCollection
             is injected as a ConcurrentCollection in ConcurrentPool. If the client's application
             wants to use and inject MCQC, it needs to provide the below dependency. -->
        <dependency>
            <groupId>com.conversantmedia</groupId>
            <artifactId>disruptor</artifactId>
            <version>${disruptor.version}</version>
            <scope>provided</scope>
        </dependency>
```
That means it should have `<optional>true</optional>` instead of `<scope>provided</scope>`. With _provided_ scope, you tell maven (and static code analysers) that you have an always-on dependency that you assume will be provided by the end-user. But if I read your comment, it is actually an optional dependency the end-user can provide.

**Comment by simeonmalchev** [comment]
Hi @bbottema, thanks for your comment, yes that's true it should be `<optional>true</optional>` with the default compile scope. I found one more of the same kind in the vibur-dbcp project pom file, it also needs to change. I can't remember why I have done it this way back at the time.. Is this provided dependency causing some immediate problem for you at the moment? I mean, I'll keep the issue open and I'll definitely fix it in future, but I wasn't planing to push out a new release right now.

**Comment by bbottema** [comment]
Nope, no issues at the moment. Just I let you know :cheers:

**Comment by simeonmalchev** [comment]
Fixed in version 24.0.
